lib/views/pages/home/content/map/components/map_pin_pop_up.dart

```diff
@@ -0,0 +1,74 @@
+import "package:flutter/material.dart";
+
+class MapPinPopUp extends StatelessWidget {
+  //key for the arrow button
+  static const arrowButtonKey = Key("arrowButton");
+  const MapPinPopUp({
+    super.key,
+    this.title,
+    this.content,
+    this.navigationAction,
+    this.displayButton = false,
+  });
+
+  final String? title;
+  final String? content;
+  final void Function()? navigationAction;
+  final bool displayButton;
```

Comment:
It could be possible that someone cretes this pop up with a navigation action but forgetting to set `displayButton` to true; and it would be possible to have a button that does nothing. I recommend to instead encoode the `displayButton` as whether `navigationAction` is null or not. If this function is null, you don't show the button, if it's not you show it.

Reply:
Fixed with [994d03c](https://github.com/ProximaEPFL/proxima/pull/335/commits/994d03c1af074c06ac3ec0bd4f9b9c8c5ff4534d)